aws-logs-metricfilter/aws-logs-metricfilter.json

```diff
@@ -0,0 +1,114 @@
+{
+  "typeName": "AWS::Logs::MetricFilter",
+  "description": "Specifies a metric filter that describes how CloudWatch Logs extracts information from logs and transforms it into Amazon CloudWatch metrics.",
+  "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs.git",
+  "definitions": {
+    "MetricTransformation": {
+      "type": "object",
+      "properties": {
+        "DefaultValue": {
+          "description": "The value to emit when a filter pattern does not match a log event. This value can be null.",
+          "type": "number"
+        },
+        "MetricName": {
+          "description": "The name of the CloudWatch metric. Metric name must be in ASCII format.",
+          "type": "string",
+          "minLength": 1,
+          "maxLength": 255,
+          "pattern": "^((?![:*$])[\\x00-\\x7F]){1,255}\\Z"
+        },
+        "MetricNamespace": {
+          "$comment": "Namespaces can be up to 256 characters long; valid characters include 0-9A-Za-z.-_/#.",
+          "description": "The namespace of the CloudWatch metric.",
+          "type": "string",
+          "minLength": 1,
+          "maxLength": 256,
+          "pattern": "^[0-9a-zA-Z\\.-_\\/#\\.]{1,256}\\Z"
+        },
+        "MetricValue": {
+          "description": "The value to publish to the CloudWatch metric when a filter pattern matches a log event.",
+          "type": "string",
+          "minLength": 1,
+          "maxLength": 100
+        }
+      },
+      "required": [
+        "MetricName",
+        "MetricNamespace",
+        "MetricValue"
+      ],
+      "additionalProperties": false
+    }
+  },
+  "properties": {
+    "FilterName": {
+      "description": "A name for the metric filter.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 512,
+      "pattern": "^[^:*]{1,512}\\Z"
+    },
+    "FilterPattern": {
+      "description": "Pattern that Logs follows to interpret each entry in a log.",
+      "type": "string",
+      "maxLength": 1024
+    },
+    "LogGroupName": {
+      "description": "Existing log group that you want to associate with this filter.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 512,
+      "pattern": "^[.\\-_/#A-Za-z0-9]{1,512}\\Z"
+    },
+    "MetricTransformations": {
+      "description": "A collection of information that defines how metric data gets emitted.",
+      "type": "array",
+      "minItems": 1,
+      "maxItems": 1,
+      "items": {
+        "$ref": "#/definitions/MetricTransformation"
+      }
+    }
+  },
+  "handlers": {
+    "create": {
+      "permissions": [
+        "logs:PutMetricFilter"
+      ]
+    },
+    "read": {
+      "permissions": [
+        "logs:DescribeMetricFilters"
+      ]
+    },
+    "update": {
+      "permissions": [
+        "logs:PutMetricFilter"
+      ]
+    },
+    "delete": {
+      "permissions": [
+        "logs:DeleteMetricFilter"
```

Comment:
[none of these are valid IAM permissions](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazoncloudwatch.html#amazoncloudwatch-actions-as-permissions)